Here are the logs from the failing step:

```
-- Configuring x64-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: /ci/myagent/_work/2/s/buildtrees/libspatialite/src-x64-linux-release/libspatialite-4.3.0a/configure --prefix=/ci/myagent/_work/2/s/buildtrees/libspatialite/src-x64-linux-release/libspatialite-4.3.0a/../../make-build-x64-linux-release "CFLAGS=-I/ci/myagent/_work/2/s/installed/x64-linux/include  " LDFLAGS=-L/ci/myagent/_work/2/s/installed/x64-linux/lib "LIBS=-lpthread -ldl" --with-geosconfig=/ci/myagent/_work/2/s/buildtrees/libspatialite/src-x64-linux-release/libspatialite-4.3.0a/geos-config "LIBXML2_LIBS=-lxml2 -llzma" LIBXML2_CFLAGS=/ci/myagent/_work/2/s/installed/x64-linux/include
    Working Directory: /ci/myagent/_work/2/s/buildtrees/libspatialite/src-x64-linux-release/libspatialite-4.3.0a
    Error code: 1
    See logs for more information:
      /ci/myagent/_work/2/s/buildtrees/libspatialite/config-x64-linux-rel-out.log
      /ci/myagent/_work/2/s/buildtrees/libspatialite/config-x64-linux-rel-err.log
```
[config-x64-linux-rel-err.log](https://github.com/microsoft/vcpkg/files/3597600/config-x64-linux-rel-err.log)
[config-x64-linux-rel-out.log](https://github.com/microsoft/vcpkg/files/3597601/config-x64-linux-rel-out.log)

--- a/ports/libspatialite/CONTROL
+++ b/ports/libspatialite/CONTROL
@@ -1,5 +1,5 @@
 Source: libspatialite
-Version: 4.3.0a-3
+Version: 4.3.0a-4
 Homepage: https://www.gaia-gis.it/gaia-sins/libspatialite-sources
 Description: SpatiaLite is an open source library intended to extend the SQLite core to support fully fledged Spatial SQL capabilities.
 Build-Depends: libxml2, sqlite3, geos, proj4, zlib, freexl, libiconv

--- a/ports/libspatialite/fix-configure-debug.patch
+++ b/ports/libspatialite/fix-configure-debug.patch
@@ -1,5 +1,5 @@
 diff --git a/configure b/configure
-index aafdad5..2715ddc 100644
+index aafdad55b..11dde35fa 100644
 --- a/configure
 +++ b/configure
 @@ -18089,7 +18089,7 @@ return GEOSCoveredBy ();
@@ -20,3 +20,39 @@ index aafdad5..2715ddc 100644
  
    #-----------------------------------------------------------------------
    #   --enable-controlpoints
+@@ -18176,7 +18176,7 @@ return GEOSDelaunayTriangulation ();
+   return 0;
+ }
+ _ACEOF
+-for ac_lib in '' geos_c; do
++for ac_lib in '' geos_cd; do
+   if test -z "$ac_lib"; then
+     ac_res="none required"
+   else
+@@ -18240,7 +18240,7 @@ fi
+ done
+ 
+ 	  _save_libs="$LIBS"
+-	  LIBS=$LIBS' -llwgeom -lgeos_c'
++	  LIBS=$LIBS' -llwgeom -lgeos_cd'
+ 	  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+ /* end confdefs.h.  */
+ int main() { return lwgeom_make_valid(); return 0; }
+@@ -18255,7 +18255,7 @@ fi
+ rm -f core conftest.err conftest.$ac_objext \
+     conftest$ac_exeext conftest.$ac_ext
+ 	  LIBS="$_save_libs"
+-	  LIBS=$LIBS' -llwgeom -lgeos_c'
++	  LIBS=$LIBS' -llwgeom -lgeos_cd'
+ 	#
+ 	# testing for PostGIS 2.1 - lwgeom_set_handlers
+ 	#
+@@ -18283,7 +18283,7 @@ return lwgeom_set_handlers ();
+   return 0;
+ }
+ _ACEOF
+-for ac_lib in '' geos_c; do
++for ac_lib in '' geos_cd; do
+   if test -z "$ac_lib"; then
+     ac_res="none required"
+   else

--- a/ports/libspatialite/portfile.cmake
+++ b/ports/libspatialite/portfile.cmake
@@ -7,12 +7,11 @@ vcpkg_download_distfile(ARCHIVE
 )
 
 if (VCPKG_TARGET_IS_WINDOWS)
-	set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/libspatialite-${LIBSPATIALITE_VERSION_STR})
-	vcpkg_extract_source_archive(${ARCHIVE})
 	find_program(NMAKE nmake)
 
-	vcpkg_apply_patches(
-		SOURCE_PATH ${SOURCE_PATH}
+	vcpkg_extract_source_archive_ex(
+		OUT_SOURCE_PATH SOURCE_PATH
+		ARCHIVE ${ARCHIVE}
 		PATCHES
 			${CMAKE_CURRENT_LIST_DIR}/fix-makefiles.patch
 			${CMAKE_CURRENT_LIST_DIR}/fix-sources.patch
@@ -115,18 +114,6 @@ if (VCPKG_TARGET_IS_WINDOWS)
 
 	message(STATUS "Packaging ${TARGET_TRIPLET} done")
 elseif (VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX) # Build in UNIX
-  # Extract source into architecture specific directory, because libspatialites' nmake based build currently does not
-  # support out of source builds.
-  set(SOURCE_PATH_DEBUG   ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-debug/libspatialite-${LIBSPATIALITE_VERSION_STR})
-  set(SOURCE_PATH_RELEASE ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-release/libspatialite-${LIBSPATIALITE_VERSION_STR})
-  
-  if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
-  	list(APPEND BUILD_TYPES "release")
-  endif()
-  if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
-  	list(APPEND BUILD_TYPES "debug")
-  endif()
-
   # Check build system first
   find_program(MAKE make)
   if (NOT MAKE)
@@ -137,23 +124,29 @@ elseif (VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX) # Build in UNIX
     ################
     # Release build
     ################    
-	file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-release)
-  	vcpkg_extract_source_archive(${ARCHIVE} ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-release)
-  	vcpkg_apply_patches(
-  		SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-release/libspatialite-${LIBSPATIALITE_VERSION_STR}
-  		PATCHES
-  		${CMAKE_CURRENT_LIST_DIR}/fix-sources.patch
-  		${CMAKE_CURRENT_LIST_DIR}/fix-latin-literals.patch
-  	)
+	vcpkg_extract_source_archive_ex(
+		OUT_SOURCE_PATH SOURCE_PATH_RELEASE
+		ARCHIVE ${ARCHIVE}
+		REF  release
+		PATCHES
+			${CMAKE_CURRENT_LIST_DIR}/fix-sources.patch
+			${CMAKE_CURRENT_LIST_DIR}/fix-latin-literals.patch
+	)
     message(STATUS "Configuring ${TARGET_TRIPLET}-rel")
     set(OUT_PATH_RELEASE ${SOURCE_PATH_RELEASE}/../../make-build-${TARGET_TRIPLET}-release)
+	file(REMOVE_RECURSE ${OUT_PATH_RELEASE})
     file(MAKE_DIRECTORY ${OUT_PATH_RELEASE})
     set(prefix ${CURRENT_INSTALLED_DIR})
     set(exec_prefix ${prefix}/bin)
     set(includedir ${prefix}/include)
     set(libdir ${prefix}/lib)
-    configure_file(${CMAKE_CURRENT_LIST_DIR}/geos-config.in
+	configure_file(${CMAKE_CURRENT_LIST_DIR}/geos-config.in
                    ${SOURCE_PATH_RELEASE}/geos-config @ONLY)
+	vcpkg_execute_required_process(
+      COMMAND chmod -R 777 ${SOURCE_PATH_RELEASE}/geos-config
+      WORKING_DIRECTORY ${SOURCE_PATH_RELEASE}
+      LOGNAME config-${TARGET_TRIPLET}-rel
+    )
     vcpkg_execute_required_process(
       COMMAND "${SOURCE_PATH_RELEASE}/configure" --prefix=${OUT_PATH_RELEASE} "CFLAGS=-I${includedir} ${VCPKG_C_FLAGS} ${VCPKG_C_FLAGS_RELEASE}" "LDFLAGS=-L${libdir}" "LIBS=-lpthread -ldl" "--with-geosconfig=${SOURCE_PATH_RELEASE}/geos-config" "LIBXML2_LIBS=-lxml2 -llzma" "LIBXML2_CFLAGS=${includedir}"
       WORKING_DIRECTORY ${SOURCE_PATH_RELEASE}
@@ -174,17 +167,11 @@ elseif (VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX) # Build in UNIX
       LOGNAME make-install-${TARGET_TRIPLET}-release
     )
 
-    set(prefix ${CURRENT_INSTALLED_DIR})
-    set(exec_prefix ${prefix}/bin)
-    set(libdir ${prefix}/lib)
-    set(includedir ${prefix}/include)
     set(VERSION ${LIBSPATIALITE_VERSION_STR})
     configure_file(${SOURCE_PATH_RELEASE}/spatialite.pc.in
-                   ${OUT_PATH_RELEASE}/share/libspatialite/pkgconfig/spatialite.pc @ONLY)
-    file(REMOVE_RECURSE ${OUT_PATH_RELEASE}/lib/pkgconfig)
+                   ${OUT_PATH_RELEASE}/lib/pkgconfig/spatialite.pc @ONLY)
     file(COPY ${OUT_PATH_RELEASE}/lib DESTINATION ${CURRENT_PACKAGES_DIR})
     file(COPY ${OUT_PATH_RELEASE}/include DESTINATION ${CURRENT_PACKAGES_DIR})
-    file(COPY ${OUT_PATH_RELEASE}/share DESTINATION ${CURRENT_PACKAGES_DIR})
     file(INSTALL ${SOURCE_PATH_RELEASE}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/libspatialite RENAME copyright)
     message(STATUS "Installing ${TARGET_TRIPLET}-rel done")
   endif()
@@ -193,17 +180,18 @@ elseif (VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX) # Build in UNIX
     ################
     # Debug build
     ################    
-	file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-debug)
-	vcpkg_extract_source_archive(${ARCHIVE} ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-debug)
-	vcpkg_apply_patches(
-		SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-debug/libspatialite-${LIBSPATIALITE_VERSION_STR}
+	vcpkg_extract_source_archive_ex(
+		OUT_SOURCE_PATH SOURCE_PATH_DEBUG
+		ARCHIVE ${ARCHIVE}
+		REF  debug
 		PATCHES
-		${CMAKE_CURRENT_LIST_DIR}/fix-sources.patch
-		${CMAKE_CURRENT_LIST_DIR}/fix-latin-literals.patch
-		${CMAKE_CURRENT_LIST_DIR}/fix-configure-debug.patch
+			${CMAKE_CURRENT_LIST_DIR}/fix-sources.patch
+			${CMAKE_CURRENT_LIST_DIR}/fix-latin-literals.patch
+			${CMAKE_CURRENT_LIST_DIR}/fix-configure-debug.patch
 	)
     message(STATUS "Configuring ${TARGET_TRIPLET}-dbg")
     set(OUT_PATH_DEBUG ${SOURCE_PATH_DEBUG}/../../make-build-${TARGET_TRIPLET}-debug)
+	file(REMOVE_RECURSE ${OUT_PATH_DEBUG})
     file(MAKE_DIRECTORY ${OUT_PATH_DEBUG})
 	set(prefix ${CURRENT_INSTALLED_DIR})
     set(exec_prefix ${prefix}/debug/bin)
@@ -211,6 +199,11 @@ elseif (VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX) # Build in UNIX
     set(libdir ${prefix}/debug/lib)
     configure_file(${CMAKE_CURRENT_LIST_DIR}/geos-config-debug.in
                    ${SOURCE_PATH_DEBUG}/geos-config @ONLY)
+	vcpkg_execute_required_process(
+      COMMAND chmod -R 777 ${SOURCE_PATH_DEBUG}/geos-config
+      WORKING_DIRECTORY ${SOURCE_PATH_DEBUG}
+      LOGNAME config-${TARGET_TRIPLET}-debug
+    )
     vcpkg_execute_required_process(
       COMMAND "${SOURCE_PATH_DEBUG}/configure" --prefix=${OUT_PATH_DEBUG}  "CFLAGS=-I${includedir} ${VCPKG_C_FLAGS} ${VCPKG_C_FLAGS_DEBUG}" "LDFLAGS=-L${libdir}" "LIBS=-lpthread -ldl" "--with-geosconfig=${SOURCE_PATH_DEBUG}/geos-config" "LIBXML2_LIBS=-lxml2 -llzmad" "LIBXML2_CFLAGS=${includedir}"
       WORKING_DIRECTORY ${SOURCE_PATH_DEBUG}
@@ -231,7 +224,9 @@ elseif (VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX) # Build in UNIX
       LOGNAME make-install-${TARGET_TRIPLET}-debug
     )
 
-    file(REMOVE_RECURSE ${OUT_PATH_DEBUG}/lib/pkgconfig)
+    set(VERSION ${LIBSPATIALITE_VERSION_STR})
+    configure_file(${SOURCE_PATH_DEBUG}/spatialite.pc.in
+                   ${OUT_PATH_DEBUG}/lib/pkgconfig/spatialite.pc @ONLY)
     file(COPY ${OUT_PATH_DEBUG}/lib DESTINATION ${CURRENT_PACKAGES_DIR}/debug)
     message(STATUS "Installing ${TARGET_TRIPLET}-dbg done")
   endif()

